File: src/main/java/com/wanted/preonboarding/domain/reservation/repository/ReservationRepository.java
package com.wanted.preonboarding.domain.reservation.repository;

import java.util.UUID;

import org.springframework.data.jpa.repository.JpaRepository;

import com.wanted.preonboarding.domain.reservation.domain.entity.Reservation;

public interface ReservationRepository extends JpaRepository<Reservation, Long> {

	boolean existsByPerformanceIdAndHallSeatId(
		UUID performanceId,
		Long performanceHallSeatId);
}

File: src/main/kotlin/com/wanted/preonboarding/ticket/infra/entity/ReservationRepository.kt
package com.wanted.preonboarding.ticket.infra.entity

import org.springframework.data.jpa.repository.JpaRepository
import java.util.UUID

interface ReservationRepository : JpaRepository<ReservationEntity, Long> {
    fun findByPerformanceId(performanceId: UUID): List<ReservationEntity>

    fun findAllByPerformanceIdIn(performanceIds: List<UUID>): List<ReservationEntity>
}
